config/configcompression/compressiontype.go

```diff
@@ -25,19 +37,46 @@ func (ct *Type) IsCompressed() bool {
 	return *ct != typeEmpty && *ct != typeNone
 }
 
-func (ct *Type) UnmarshalText(in []byte) error {
-	typ := Type(in)
-	if typ == TypeGzip ||
-		typ == TypeZlib ||
-		typ == TypeDeflate ||
-		typ == TypeSnappy ||
-		typ == TypeZstd ||
-		typ == TypeLz4 ||
-		typ == typeNone ||
-		typ == typeEmpty {
-		*ct = typ
+func (ct *TypeWithLevel) UnmarshalText(in []byte) error {
```

Comment:
As mentioned in https://github.com/open-telemetry/opentelemetry-collector/pull/11084#issuecomment-2428122545, I would recommend splitting all validation out into a separate `func (*TypeWithLevel) Validate() error` method.

Reply:
I took the approach of having all the validation in `UnmarshalText` based on this previous review comment by @dmitryax  - https://github.com/open-telemetry/opentelemetry-collector/pull/11084#discussion_r1776033836

Reply:
Continuing discussion on that thread.